src/index.ts

```diff
@@ -1,228 +1,342 @@
 #!/usr/bin/env node
 
 import { intro, text, tasks, outro, isCancel, cancel, log, note } from '@clack/prompts';
-import { exec, execSync } from 'node:child_process';
+import { existsSync } from 'node:fs';
+import { exec } from 'node:child_process';
 import { promises } from 'fs';
 import { eslintConfig, prettierConfig } from './config/linting';
 import { vsCodeSettings } from './config/vsCode';
 import { gitIgnore } from './config/git';
 import { jestConfig } from './config/jest';
 import path from 'node:path';
+import { notes, readme, sum, sumTest, techdebt } from './config/initialCode';
 
-const sum = `export function sum(a: number, b: number): number {
-  return a + b;
-}
-`
-
-const sumTest = `import { sum } from '@/index';
-
-describe('Sum should be', () => {
-  test('total of 3 when adding 1 and 2', () => {
-    // arrange
-    const expectedOutput = 3;
-
-    // act
-    let result = sum(1, 2);
-
-    // assert
-    expect(result).toBe(expectedOutput);
-  });
-});
-`
-
-const notes = `✅ DONE
-
-⚠️ TODO
-
-🚧 WIP
-
-🅿️ PARKED
-`
-
-const readme = `# README
-
-A story about your amazing kata excersize here.`
-
-const techdebt = ``
+const CURRENT_DOJO = 'ns_white_crane_white_belt';
 
 function onCancel(message: string) {
-    cancel(message);
-    process.exit(0);
+  cancel(message);
+  process.exit(0);
 }
 
 function myExec(cmd: string): Promise<string> {
-    return new Promise((resolve) => {
-        myExecImpl(resolve, cmd)
-    });
+  return new Promise((resolve) => {
+    myExecImpl(resolve, cmd);
+  });
 }
 
 function myExecImpl(callback: (output: string) => void, cmd: string) {
-    exec(cmd, (error, stdout, stderr) => {
-        if (error) throw error;
-        callback(stdout || stderr);
-    })
+  exec(cmd, (error, stdout, stderr) => {
+    if (error) throw error;
+    callback(stdout || stderr);
+  });
```

Comment:
Instead of throwing, return it in the callback so it can be rejected to the promise.
```suggestion
  exec(cmd, (error, stdout, stderr) => {
    if (error) callback(stderr, error);
    callback(stdout || stderr, undefined);
  });
```